File: Java/List.java
public class <T> List{
  private int size = 0 , capacity = 5;
  private Object arr[];

  List(int maxSize){
    this.arr = new T[maxSize];
    this.capacity = maxSize;
  }

  List(){
    this.arr = new Object[this.maxSize];
  }

  public void append(T data){
    if(this.size >= this.capacity) {
      this.resize();
    }
    this.arr[size] = data;
    this.size += 1;
  }

  public void resize(){
    this.capacity += this.capacity;
    newArr = new Object[this.capacity];
    for(int i = 0; i < this.size; i++){
      newArr[i] = this.arr[i];
    }

    this.arr = newArr;
  }

  public void insert(int pos, T data){
    if(this.size >= this.capacity){
      this.resize();
    }

    T temp = data;
    T temp2;

    for(int i = pos; i <= this.size; i++){
      temp2 = this.arr[i];
      this.arr[i] = temp;
      temp = temp2;
    }
  }


  public static void main(String args[]) {
    
  }

}

File: C++/List.cpp
#include<iostream>
#include "stdio.h"

template <typename T>
class List{
  private:
    int size, capacity;
    T arr[];

  public:
    List(){
      size = 0;
      capacity = 5;
      arr = new T[5];
    }

    List(int max_size){
      size = 0;
      capacity = max_size;
      arr = new T[max_size];
    }

    void append(T data){
      if(size >= capacity){
        resize();
      }
      arr[size] = data;
      size++;
    }

    void resize(){
      capacity += capacity;
      T new_arr[] = new T[capacity];
      for(int i = 0; i < size; i++){
        new_arr = arr[i];
      }

      T* temp = arr;
      arr = new_arr;
      delete temp;
    }

    void display(){
      std::cout << " { ";
      for(T i : arr){
        std::cout << i << " , ";
      }
      std::cout << " } ";
    }

};

int main(){
  List<int>* lst = new List<int>();
  for(int i = 0; i < 5; i++){
    (*lst).append(i);
  }
  (*lst).display();
}

